flap/plot.py

```diff
@@ -311,204 +297,37 @@ def animate(self):
             
             self.xdata_range=None
             self.ydata_range=None
+            if (self.xrange is None):
+                self.xrange=[np.min(self.xdata),np.max(self.xdata)]
+            
+            plt.xlim(self.xrange[0]*self.axes_unit_conversion[0],
+                     self.xrange[1]*self.axes_unit_conversion[0])
+                
+            if (self.yrange is None):
+                self.yrange=[np.min(self.ydata),np.max(self.ydata)]
+                
+            plt.ylim(self.yrange[0]*self.axes_unit_conversion[1],
+                     self.yrange[1]*self.axes_unit_conversion[1]) 
 
         if (self.options['Colorbar']):
             cbar = plt.colorbar(img,ax=self.ax_act)
             cbar.set_label(self.d.data_unit.name)
 #EFIT overplot feature implementation:
             #It needs to be more generalied in the future as the coordinates are not necessarily in this order: [time_index,spat_index]
-            #This needs to be cross-checked with the time array's dimensions wherever there is a call for a certain index.
-        if ('EFIT options' in self.options and self.options['EFIT options'] is not None):
```

Comment:
This part was removed because it is too specific for the EFIT. It has been merged into the overplotting options.